packages/react-native-web/src/exports/TextInput/index.js

```diff
@@ -332,9 +332,20 @@ class TextInput extends Component<*> {
   };
 
   _handleKeyPress = e => {
-    const { blurOnSubmit, multiline, onKeyPress, onSubmitEditing } = this.props;
+    const {
+      blurOnSubmit,
+      multiline,
+      maxLength,
+      keyboardType,
+      onKeyPress,
+      onSubmitEditing
+    } = this.props;
     const blurOnSubmitDefault = !multiline;
     const shouldBlurOnSubmit = blurOnSubmit == null ? blurOnSubmitDefault : blurOnSubmit;
+    if (keyboardType === 'numeric' && e.target.value.length >= maxLength) {
```

Comment:
`preventDefault` is cancelling all key events here. To apply this code only when the key pressed is a number, we might need something more like this (but I don't know how reliable the condition in the suggestion is)

```suggestion
    if (keyboardType === 'numeric' && e.target.value.length >= maxLength && !isNaN(parseInt(e.key, 10))) {
```

Reply:
it will break for floating point number `1e5`

Reply:
e.key is never going to be that value